lib/gitsh/completer.rb

```diff
@@ -37,11 +37,30 @@ def available_completers
       end
 
       def completing_arguments?
-        full_input = readline.line_buffer
         tokens = full_input.split
         tokens.any? && full_input.end_with?(' ') || tokens.size > 1
       end
 
+      def completing_quoted_argument?
```

Comment:
This could probably be more accurately implemented using `rl_completion_found_quote`, `rl_completion_quote_character`, or some combination of the two.